qutebrowser/app.py

```diff
@@ -814,6 +826,22 @@ def on_focus_object_changed(self, obj):
             log.misc.debug("Focus object changed: {}".format(output))
         self._last_focus_object = output
 
+    @pyqtSlot(Qt.ApplicationState)
+    def on_app_state_changed(self, state):
+        if state != Qt.ApplicationActive:
+            return
+
+        window = objreg.last_focused_window()
+        mainwindow.raise_window(window)
+
+    def event(self, e):
+        if e.type() == QEvent.FileOpen:
+            open_url(e.url(), force_raise=False)
+        else:
+            return super().event(e)
```

Comment:
With the `return` above, you can get rid of the else ([related lightning talk 😆](https://www.youtube.com/watch?v=JVVMMULwR4s&feature=youtu.be&t=4m57s))

Reply:
Yes, this is what I usually do (removing superfluous else). But if you look at this method as a whole (+ consider your previous comment), you'll see what's going on here :) It's an insurance to always `return True` for custom handlers (I've been bitten by this few times).

```python
    def event(self, e):
        if e.type() == QEvent.FileOpen:
            open_url(e.url(), force_raise=False)
        else:
            # events, not handled by this method, returns here
            return super().event(e)

        # events, being handled by this method, always end up here
        return True
```

If you think it's not needed, I'm absolutely ok to remove this.

Reply:
Ah, right, I missed the `return True` - sorry! Seems fine to keep as-is.